### PR TITLE
Added Starting block

### PR DESCRIPTION
**Pull request by KatAttack01**
I think the levels might make merge conflicts. How do you not include those and just add the prefabs only instead?

### CODE REVIEW

File: Assets/Scripts/PlayerMovement/CatMovement.cs [hunk 1]
```diff
@@ -388,10 +390,19 @@ IEnumerator SetObject()
...
         blockManager = map.GetComponentInChildren<BlockManager>();
+        try
+        {
+            transform.position = map.GetComponentInChildren<StartingBlock>().GetPos();
+        }
+        catch
+        {
+            Debug.Log("starting block not made");
+            transform.position = new Vector3(0, 0, 0);
+        }
```
Comment (Andrew-Ting) [inline Assets/Scripts/PlayerMovement/CatMovement.cs:404]: Honestly not a fan of try catch statements. They catch all errors, and they tend to nest fast if you want to catch specific errors. Imo they're best used only when a certain line of code can cause a multitude of problems and it's impossible to predict/handle every one (e.g. database queres, API requests, etc.)

In this case you do know what the problem is. If you want to add a guard for the addition of the starting block, I'd say just be explicit and check if `map.GetComponentInChildren<StartingBlock>()` is defined before setting `transform.position` to its `GetPos` function.

**Review by Andrew-Ting — CHANGES_REQUESTED** [review]
I seem to have missed it in a previous PR, but can you change `SetObject` to something like `InitializeCatReferences` instead? Just a personal preference, but I like naming only getters with the prefix Get and setters with the prefix Set. 

The naming scheme is nice since it allows you to type "get/set" after selecting a script and see what info you can take/set in another script without worrying about doing anything beyond what the name suggests. It makes code reusability easier.

File: Assets/Scripts/PlayerMovement/CatMovement.cs [hunk 2]
```diff
@@ -18,7 +18,9 @@ public class CatMovement : MonoBehaviour
     [SerializeField]
     LayerMask objects = (1 << 6 | 1 << 7);
     public event Action<Vector3> CatMoveAction;
+    public event Action<bool> CatLoaded;
```
[the remark this answers is not in the document]
Comment (KatAttack01) [reply Assets/Scripts/PlayerMovement/CatMovement.cs:21]: CatLoaded is for camera script I think

File: Assets/Scripts/PlayerMovement/CatMovement.cs [hunk 3]
```diff
@@ -388,10 +390,21 @@ IEnumerator SetObject()
...
+        {
+            //it doesn't exist
+            Debug.Log("starting block not made");
+            transform.position = Vector3.zero;
+        }
+        else
+        {
+
+        }
+            transform.position = map.GetComponentInChildren<StartingBlock>().GetPos();
```
Comment (Andrew-Ting) [inline Assets/Scripts/PlayerMovement/CatMovement.cs:406]: ? shouldn't this be in the else statement?

File: Assets/Scripts/PlayerMovement/CatMovement.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (Andrew-Ting) [reply Assets/Scripts/PlayerMovement/CatMovement.cs]: Gotcha

**Review by Andrew-Ting — CHANGES_REQUESTED** [review]

File: Assets/Scripts/PlayerMovement/CatMovement.cs (hunk 3 above)
Comment (KatAttack01) [reply Assets/Scripts/PlayerMovement/CatMovement.cs:406]: oh right, it should be haha

**Review by Andrew-Ting — APPROVED** [review]
👍

**Pull request merged**